src/services/fileManager.ts

```diff
@@ -245,6 +247,44 @@ export async function recalculateFileState(file: File): Promise<File> {
   return file;
 }
 
+/**
+ * Check with clinical service if the file has had clinical data added, update the file, save to the DB
+ * and return the file with any changes.
+ * @param file
+ * @returns
+ */
+export async function updateFileClinicalData(file: File): Promise<File> {
+  if (file.hasClinicalData) {
+    // File already known to have clinical data, no changes necessary
+    return file;
+  }
+
+  try {
+    const donorData = await clinical.getDonor(file.programId, file.donorId);
+    if (donorData?.completionStats?.coreCompletionPercentage === 1) {
+      return await fileService.updateFileHasClinicalData(file.objectId, true);
+    }
+  } catch (error) {
+    logger.warn(`Failed to update file clinical data record`, file.programId, file.donorId, error);
+  }
+  return file;
+}
```

Comment:
This function is the important new work. It fetches donor data from clinical and updates the file `hasClinicalData` property to reflect the clinical data state.

This is skipped if the file is already marked with `hasClinicalData` under the assumption that clinical data won't be removed.

Reply:
Is there a possible case for a file that has clinical data, for it to go "missing"?
e.g. the clinical data gets removed for some valid reason (embargo of some sort, I dunno the legalese/jargon).